pkg/build/api/validation/validation.go

```diff
@@ -241,6 +245,9 @@ func validateGitSource(git *buildapi.GitBuildSource) fielderrors.ValidationError
 	if len(git.HTTPSProxy) != 0 && !isValidURL(git.HTTPSProxy) {
 		allErrs = append(allErrs, fielderrors.NewFieldInvalid("httpsproxy", git.HTTPSProxy, "proxy is not a valid url"))
 	}
+	if hasProxy(git) && !strings.HasPrefix(git.URI, "http") {
```

Comment:
parse the URL and check the scheme... I always clone ssh repos as `httpuser@myrepo.com` :)